<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1226</width>
    <height>804</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Smoothing out slice-slice intensity fluctuations</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="pushButtonInputTiff">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Input tiff filename</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>730</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Result:</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelResult">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>730</y>
      <width>491</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>10</y>
      <width>341</width>
      <height>221</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelOutputTiffFile">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>30</y>
      <width>761</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="labelInputTiffFile">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>761</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonRead">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>90</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Read input tiff</string>
    </property>
   </widget>
   <widget class="QwtPlot" name="qwtPlot_frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>861</width>
      <height>541</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonOutputTiff">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Output tiff filename</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_threshold">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>90</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>10</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_threshold">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>90</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Threshold</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>60</y>
      <width>71</width>
      <height>81</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <widget class="QRadioButton" name="radioButton_X">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>61</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>X slices</string>
     </property>
     <attribute name="buttonGroup">
      <string>buttonGroupAxis</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="radioButton_Y">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>61</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Y slices</string>
     </property>
     <attribute name="buttonGroup">
      <string>buttonGroupAxis</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="radioButton_Z">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>61</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Z slices</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string>buttonGroupAxis</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButtonSmooth">
    <property name="geometry">
     <rect>
      <x>494</x>
      <y>90</y>
      <width>51</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Smooth</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_smoothing">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>90</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>0.05</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_parameter">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>90</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Fraction</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonSum">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>90</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Sum intensity</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonScale">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>90</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonSave">
    <property name="geometry">
     <rect>
      <x>764</x>
      <y>90</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Write output tiff</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1226</width>
     <height>18</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonInputTiff</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>tiffInFileSelecter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>56</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonOutputTiff</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>tiffOutFileSelecter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonRead</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>readRawData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>590</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonSmooth</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>smoother()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>517</x>
     <y>138</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonGroupAxis</sender>
   <signal>buttonClicked(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>axisChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>612</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonSum</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sumIntensity()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_threshold</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>thresholdChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>427</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonScale</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>determineScale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>804</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonSave</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveScaledTiff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>790</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>751</x>
     <y>90</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>readRawData()</slot>
  <slot>tiffInFileSelecter()</slot>
  <slot>tiffOutFileSelecter()</slot>
  <slot>smoother()</slot>
  <slot>axisChanged()</slot>
  <slot>sumIntensity()</slot>
  <slot>thresholdChanged()</slot>
  <slot>determineScale()</slot>
  <slot>saveScaledTiff()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroupAxis"/>
 </buttongroups>
</ui>
